incubator/binding-mqtt-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/mqtt/kafka/streams/kafka/session.will.message.abort.deliver.will.retain/server.rpt

```diff
@@ -0,0 +1,380 @@
+                               .header("type", "will-signal")
+                               .build()
+                           .build()}
+
+read zilla:data.empty
+
+read zilla:data.ext ${kafka:matchDataEx()
+                           .typeId(zilla:id("kafka"))
+                           .merged()
+                               .deferred(0)
+                               .partition(-1, -1)
+                               .key("client-1#will")
+                               .hashKey("client-1")
+                               .build()
+                           .build()}
+
+read zilla:data.empty
+
+# non-clean start -> fetch lifetimeId
+accepted
+
+read zilla:begin.ext ${kafka:matchBeginEx()
+                            .typeId(zilla:id("kafka"))
+                            .merged()
+                                .capabilities("PRODUCE_AND_FETCH")
+                                .topic("mqtt_sessions")
+                                .filter()
+                                    .key("client-1#will-signal")
+                                    .build()
+                                .build()
+                            .build()}
+
+connected
+
+write advise zilla:flush
+
+read closed
+write close
+
+
+# send initial migrate signal
+accepted
+
+read zilla:begin.ext ${kafka:matchBeginEx()
+                            .typeId(zilla:id("kafka"))
+                            .merged()
+                                .capabilities("PRODUCE_AND_FETCH")
+                                .topic("mqtt_sessions")
+                                .filter()
+                                    .key("client-1#migrate")
+                                    .headerNot("sender-id", "sender1")
+                                    .build()
+                                .build()
+                            .build()}
+
+connected
+
+read zilla:data.ext ${kafka:matchDataEx()
+                           .typeId(zilla:id("kafka"))
+                           .merged()
+                               .deferred(0)
+                               .partition(-1, -1)
+                               .key("client-1#migrate")
+                               .hashKey("client-1")
+                               .header("sender-id", "sender1")
+                               .build()
+                           .build()}
+read zilla:data.empty
+
+write notify INITIAL_MIGRATE_SENT
+
+read closed
+write close
+
+
+accepted
+
+read zilla:begin.ext ${kafka:matchBeginEx()
+                               .typeId(zilla:id("kafka"))
+                               .group()
+                                   .groupId("client-1")
+                                   .protocol("highlander")
+                                   .timeout(1000)
+                                   .build()
+                               .build()}
+
+connected
+
+# wait so that we won't be the leader too early, before the lifetimeId arrives
```

Comment:
When we become the leader, that's when we're supposed to send the will message, so that's when we're sending the window back to mqtt stream. If we become the leader early, before we could fetch the lifetimeId, it will be empty, which might be wrong.

Reply:
Understood. How does that related to _migrate_ signal though, not will message or will signal?